pkg/cri/client/client.go

```diff
@@ -202,6 +214,27 @@ func (c *client) connect(kind, socket string, options ConnectOptions) (*grpc.Cli
 	return cc, nil
 }
 
+func (c *client) dialNotify(socket string) {
+	if c.options.DialNotify == nil {
+		return
+	}
+
+	info, err := os.Stat(socket)
+	if err != nil {
+		c.options.DialNotify(socket, 0, 0, 0, err)
```

Comment:
This works and is fine, but it just feels pretty brave to set the UID and GID to "root" when we call the function with an error value :-)

Reply:
True, dangerous to trust the other end to not be brain-damaged. Changed those to -1's. Also changed the order of arguments to be more logical/natural.